### DISCUSSION TITLE
当map中定义的key为int时，decode后map会变成array

### DISCUSSION THREAD

**Discussion by averyboy** [starwing/lua-protobuf#172] [General]

**Comment by averyboy** [comment]
我在proto中定义了一个字段类型为map<int32, bool>，从pb decode成table时，map会被翻译成一个数组，{1:true} -> [true]

**Comment by starwing** [comment]
Lua里面table本来就同时是map和array

**Comment by averyboy** [comment]
【Lua里面table本来就同时是map和array】，嗯嗯，之前想到是这个原因导致的，主要是想看看有没有啥方法可以保留这个type信息，貌似只要经过lua这一层变成数组，key就找不回来了，貌似不可行。

**Comment by starwing** [comment]
其实你直接把table当map然后pairs就行了，你不需要关心它到底具体是什么的。

averyboy ***@***.***>于2021年8月9日 周一17:06写道：

>
> 【Lua里面table本来就同时是map和array】，嗯嗯，之前想到是这个原因导致的，主要是想看看有没有啥方法可以保留这个type信息，貌似只要经过lua这一层变成数组，key就找不回来了，貌似不可行。
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/starwing/lua-protobuf/issues/153#issuecomment-895062840>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA36MZ2MQF2RLS6RC6UUOLT36K7VANCNFSM5BO3OLEA>
> .
>
-- 
regards,
Xavier Wang.

**Comment by averyboy** [comment]
ok，我看看这里怎么修改下，谢谢starwing
